python/src/uagents/communication.py

```diff
@@ -104,6 +105,17 @@ async def run(self):
                     sync=sync,
                 )
                 response_future.set_result(result)
+
+                self.sent_messages.append({
+                    "timestamp": time(),
```

Comment:
Let's make that a pydantic `BaseModel` in envelope.py

```python

class EnvelopeHistoryEntry(BaseModel):
    timestamp: int = Field(default_factory=lambda: int(time.time()))
    version: int
    sender: str
    target: str
    session: UUID4
    schema_digest: str
    protocol_digest: Optional[str] = None
    payload: Optional[str] = None

    @field_serializer("session")
    def serialize_dt(self, session: UUID4, _info):
        return str(session)


class EnvelopeHistory(BaseModel):
    envelopes: List[EnvelopeHistoryEntry]
```

That way we can more easily create these:
```python
self.send_messages.append(EnvelopeHistoryEntry(**env))
```
 and when sending we just dump the whole model with:
```python
entry.model_dump_json()
```